android/KMEA/app/src/main/assets/android-host.js

```diff
@@ -337,11 +325,11 @@ function executePopupKey(keyID, keyText) {
   keyman.executePopupKey(keyID);
 }
 
-function executeHardwareKeystroke(code, shift, lstates, eventModifiers) {
+async function executeHardwareKeystroke(code, shift, lstates, eventModifiers) {
   console_debug('executeHardwareKeystroke(code='+code+',shift='+shift+',lstates='+lstates+',eventModifiers='+eventModifiers+')');
   try {
     executingHardwareKeystroke = true;
-    if (keyman.executeHardwareKeystroke(code, shift, lstates)) { // false if matched, true if not
+    if (await keyman.hardKeyboard.raiseKeyEvent(code, shift, lstates)) { // false if matched, true if not
```

Comment:
Why the change to async? Is this a breaking API change (both API endpoint rename and async change)?